File: examples/location/save/node/saveAnEntity.js
const { LocationService } = require("m3o/location");

// Save an entity's current position
async function saveAnEntity() {
  let locationService = new LocationService(process.env.MICRO_API_TOKEN);
  let rsp = await locationService.save({
    entity: {
      id: "1",
      location: {
        latitude: 51.511061,
        longitude: -0.120022,
        timestamp: "1622802761",
      },
      type: "bike",
    },
  });
  console.log(rsp);
}

saveAnEntity();

File: examples/go/location/save/saveAnEntity.go
package example

import (
	"fmt"
	"os"

	"github.com/micro/services/clients/go/location"
)

// Save an entity's current position
func SaveAnEntity() {
	locationService := location.NewLocationService(os.Getenv("MICRO_API_TOKEN"))
	rsp, err := locationService.Save(&location.SaveRequest{
		Entity: &location.Entity{
			Id: "1",
			Location: &location.Point{
				Latitude:  51.511061,
				Longitude: -0.120022,
				Timestamp: 1622802761,
			},
			Type: "bike",
		},
	})
	fmt.Println(rsp, err)
}
